packages/editor/src/components/widgets/code-editor/useCodeEditor.ts

```diff
@@ -38,8 +38,14 @@ export const useMonacoEditor = (options?: { modifyAction?: ModifyAction }) => {
         }
       ]);
     } else {
-      const text = value.cursorValue.length > 0 && options?.modifyAction ? options.modifyAction(value.cursorValue) : value.cursorValue;
+      const text =
+        value.cursorValue.length > 0 && options?.modifyAction && type !== 'catPath'
+          ? options.modifyAction(value.cursorValue)
+          : value.cursorValue;
       editor.executeEdits('browser', [{ range: selection, text, forceMoveMarkers: true }]);
+      if (type === 'catPath') {
+        editor.setValue(text);
+      }
```

Comment:
arr, a little bit ugly, but I think I understand why it is needed...
Maybe a sign that we should somehow change this or at last refactor this code a bit into smaller readable functions...